SCons/Environment.py

```diff
@@ -1045,7 +1050,12 @@ def get_CacheDir(self):
                 return self._last_CacheDir
```

Comment:
If you call CacheDir twice on the same environment with the same path, but a different custom_class, the CacheDir class will not be updated. I think I should update this to also check if CACHEDIR_CLASS is the same type as the current _last_CacheDir obj.

Reply:
resolved e0a70f1